include/parser/trees/parse_tree.h

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file parse_tree.h
+ * @author Chase Geigle
+ *
+ * All files in META are released under the MIT license. For more details,
+ * consult the file LICENSE in the root of the project.
+ */
+
+#ifndef META_PARSE_PARSE_TREE_H_
+#define META_PARSE_PARSE_TREE_H_
+
+#include <memory>
+#include "parser/trees/node.h"
+#include "parser/trees/visitors/tree_transformer.h"
+
+namespace meta
+{
+namespace parser
+{
+
+/**
+ * Represents the parse tree for a sentence. This may either be a sentence
+ * parsed from training data, or the output from a trained parser on test
+ * data.
+ *
+ * @todo determine what parts of analyzers::parse_tree are worth
+ * keeping---that class deals specifically with trees read from the output
+ * of the Stanford parser. When we have our own, we may still want some of
+ * that functionality to allow people to use parsers that are not our
+ * own?
+ */
```

Comment:
The tree format that `analyzers::parse_tree` is reading is actually just the output of meta-stanford-preprocessor. So it could be output from any parser since the format is just nested parens. Non-parens are treated as tags/labels.